src/components/Receiving/ReceivingHistory.js

```diff
@@ -2,28 +2,67 @@ import React, { Component } from 'react';
 import { FormattedMessage } from 'react-intl';
 import PropTypes from 'prop-types';
 import ReactRouterPropTypes from 'react-router-prop-types';
+import { get } from 'lodash';
 
 import {
+  Button,
+  Checkbox,
+  Col,
   IconButton,
+  MultiColumnList,
   Pane,
   PaneMenu,
   Paneset,
+  Row,
+  SearchField,
 } from '@folio/stripes/components';
 
+import {
+  ORDER,
+  RECEIVING_HISTORY as RECEIVING_HISTORY_RESOURCE,
+} from '../Utils/resources';
+import FolioFormattedTime from '../FolioFormattedTime';
 import ReceivingLinks, { RECEIVING_HISTORY } from './ReceivingLinks';
+import { PIECE_STATUS_RECEIVED } from './const';
+
+const reducePieces = (pieces, isSelect = false) => {
+  const pieceReducer = (accumulator, piece) => {
+    accumulator[piece.id] = isSelect ? piece : null;
+
+    return accumulator;
+  };
+
+  return pieces.reduce(pieceReducer, {});
+};
 
 class ReceivingHistory extends Component {
   static manifest = Object.freeze({
-    query: {
-      initialValue: {
-        query: '',
-      },
-    },
+    query: {},
+    receivingHistory: RECEIVING_HISTORY_RESOURCE,
+    order: ORDER,
   })
 
   static propTypes = {
     location: ReactRouterPropTypes.location,
     mutator: PropTypes.object,
+    resources: PropTypes.object,
```

Comment:
I'm not sure if we need to describe such thing as `mutator` and `resources` through `PropTypes.shape()`. IMHO, `PropTypes.object` is enough, but I'm open to suggestions.